src/main/java/qoobee/TaskList.java

```diff
@@ -0,0 +1,130 @@
+package qoobee;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents the list of tasks.
+ */
+public class TaskList {
+
+    private Storage storage;
+    private List<Task> taskList;
+
+    /**
+     * Creates a tasklist given a storage.
+     * @param storage The object where the tasklist will be stored to.
+     */
+    public TaskList(Storage storage) {
+        this.storage = storage;
+        this.taskList = storage.getList();
+    }
+
+    /**
+     * Returns the size of the tasklist.
+     * @return The size of the tasklist.
+     */
+    public int taskListSize() {
+        return this.taskList.size();
+    }
+
+    /**
+     * Retrieves a task given an index inputted.
+     * @param index The index of the task identified.
+     * @return Returns a task.
+     */
+    public Task getTask(int index) {
+        return this.taskList.get(index);
+    }
+
+    /**
+     * Prints the list of tasks.
+     */
+    public String printTasks() {
+        if (taskListSize() == 0) {
+            return "You have no tasks dummy!";
+        } else {
+            StringBuilder sb = new StringBuilder("Here are the tasks in your list:\n");
+            for (int i = 0; i < taskListSize(); i++) {
+                Task currentTask = taskList.get(i);
+                sb.append((i + 1) + "." + currentTask + "\n");
+            }
+            return sb.toString();
+        }
+    }
+
+    /**
+     * Adds a task into the list.
+     * @param task A task to be added in the list.
+     * @throws QoobeeException if the user inputs an invalid syntax.
+     */
+    public String addTask(Task task) throws QoobeeException {
+        this.taskList.add(task);
+        storage.save(taskList);
+        return "Got it. I've added this task:\n" + task + "\n"
+                + "Now you have " + taskListSize() + " tasks in the list.";
+    }
```

Comment:
```suggestion
    /**
     * Adds a task into the list.
     *
     * @param task A task to be added in the list.
     * @return A String with task added successfully message
     * @throws QoobeeException if the user inputs an invalid syntax.
     */
    public String addTask(Task task) throws QoobeeException {
        this.taskList.add(task);
        storage.save(taskList);
        return "Got it. I've added this task:\n" + task + "\n"
                + "Now you have " + taskListSize() + " tasks in the list.";
    }
```

I realised for most of the Javadoc comments, you have missed out on the return tag for methods with a return statement. Also, just a minor error, remember to leave an empty line between the description and the tags.